apps/etterlatte-saksbehandling-ui/client/src/components/behandling/avkorting/AvkortingInntekt.tsx

```diff
@@ -223,76 +215,65 @@ export const AvkortingInntekt = ({
       )}
       {avkortingGrunnlag.length > 1 && <TextButton isOpen={visHistorikk} setIsOpen={setVisHistorikk} />}
       {erRedigerbar && (
-        <InntektAvkortingForm onSubmit={onSubmit}>
+        <InntektAvkortingForm>
           <Rows>
             {formToggle && (
               <>
                 <FormWrapper>
                   <HStack gap="4">
                     <TextField
+                      {...register('aarsinntekt', {
+                        required: { value: true, message: 'Må fylles ut' },
+                      })}
                       label="Forventet årsinntekt Norge"
                       size="medium"
                       type="text"
                       inputMode="numeric"
                       pattern="[0-9]*"
-                      value={inntektGrunnlagForm.aarsinntekt}
-                      onChange={(e) =>
-                        setInntektGrunnlagForm({
-                          ...inntektGrunnlagForm,
-                          aarsinntekt: e.target.value === '' ? undefined : Number(e.target.value),
-                        })
-                      }
+                      error={errors.aarsinntekt?.message}
                     />
                     <TextField
+                      {...register('fratrekkInnAar', {
+                        required: { value: true, message: 'Må fylles ut' },
+                      })}
                       label="Fratrekk inn-år"
                       size="medium"
                       type="text"
                       inputMode="numeric"
                       pattern="[0-9]*"
```

Comment:
pattern kan endres til å bli validert av RHF, så har vi ikke 2 forskjellige greier for å validere ☺️

Reply:
Og på TextFieldene videre nedover